podcasts/DownloadManager.swift

```diff
@@ -459,4 +470,21 @@ class DownloadManager: NSObject, FilePathProtocol {
         downloadTask?.taskDescription = taskId
         downloadTask?.resume()
     }
+
+    func resumeAllBackgrounded() {
+        let queuedEpisodes = DataManager.sharedManager.findEpisodesWhere(customWhere: "episodeStatus == ?", arguments: [DownloadStatus.queued.rawValue])
+        queuedEpisodes.forEach { episode in
+            Task {
+                addToQueue(episodeUuid: episode.uuid)
+            }
+        }
+    }
+
+    private func existingTask(for episode: BaseEpisode) async -> URLSessionTask? {
+        let cellTasks = await cellularBackgroundSession.allTasks
+        let wifiTasks = await wifiOnlyBackgroundSession.allTasks
+
+        let cellTask = cellTasks.first(where: { $0.taskDescription == episode.downloadTaskId })
+        return cellTask ?? wifiTasks.first(where: { $0.taskDescription == episode.downloadTaskId })
```

Comment:
Can we tasks for the same episode coexist on both sessions? or it's one or the other?

Reply:
The main case I think would be if download settings were changed and an episode download was added to a different session. Similar to above, maybe we could cancel tasks which are still added to the other sessions but I feel like I should test that a bit further just to be sure.

For now, I'll change this to only check for existing tasks in the session where the download task is about to be added.